tests/transact_items_test.go

```diff
@@ -0,0 +1,122 @@
+package tests
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/oolio-group/dynago"
+)
+
+type Terminal struct {
+	Id string
+	Pk string
+	Sk string
+}
+
+func TestTransactItems(t *testing.T) {
+	endoint, purge := startLocalDatabase(t)
+	defer purge()
+
+	table := prepareTable(t, endoint, "transcact_item_test")
+
+	testCases := []struct {
+		title       string
+		condition   string
+		keys        map[string]types.AttributeValue
+		opts        []dynago.QueryOptions
+		source1     []Terminal
+		source2     []Terminal
+		expected    []Terminal
+		expectedErr error
+	}{{
+		title:     "assign terminal",
+		condition: "pk = :pk",
+		keys: map[string]types.AttributeValue{
+			":pk": &types.AttributeValueMemberS{Value: "terminal1"},
+		},
+		source1: []Terminal{
+			{
+				Id: "1",
+				Pk: "terminal1",
+				Sk: "merchant1",
+			},
+		},
+		source2: []Terminal{
```

Comment:
If you change the test case to include the list of operations instead of source1 and source 2, we can customize operations per test case here itself.

Example
![image](https://github.com/oolio-group/dynago/assets/4880359/267cf99b-78fc-4552-ade6-4b38102a4a84)

Reply:
yep addressed